dashboard/app/models/levels/blockly.rb

```diff
@@ -562,4 +563,16 @@ def shared_functions
       SharedBlocklyFunction.where(level_type: type).map(&:to_xml_fragment)
     end.join
   end
+
+  # Display translated custom block text and options
+  def localized_shared_blocks(level_object)
+    if level_object.empty?
```

Comment:
I don't know enough about blockly to review this change holistically, but I pulled your branch and if you change this line to `level_object.nil?` your failing UI tests passed for me locally.  Is it ok to return nil here in that case? Is that what you intended?

Reply:
It is ok to have nil in this case but only check for nil results in other tests failing.  I decided to check if the level object is blank because it will work on any object as well as act as .empty.